test/1_stdlib/StringDiagnostics.swift

```diff
@@ -60,20 +63,57 @@ func testStringDeprecation(hello: String) {
 
 }
 
-// Positive and negative tests for String collection types
-func acceptsCollection<I: Collection>(_: I) {}
-func acceptsBidirectionalCollection<I: BidirectionalCollection>(_: I) {}
-func acceptsRandomAccessCollection<I: RandomAccessCollection>(_: I) {}
+// Positive and negative tests for String collection types. Testing the complete
+// set of conformance just in case protocol hierarchy changes.
+func acceptsCollection<C: Collection>(_: C) {}
+func acceptsBidirectionalCollection<C: BidirectionalCollection>(_: C) {}
+func acceptsRandomAccessCollection<C: RandomAccessCollection>(_: C) {}
 
 func testStringCollectionTypes(s: String) {
   acceptsCollection(s.utf8)
   acceptsBidirectionalCollection(s.utf8) // expected-error{{argument type 'String.UTF8View' does not conform to expected type 'BidirectionalCollection'}}
+  acceptsRandomAccessCollection(s.utf8) // expected-error{{argument type 'String.UTF8View' does not conform to expected type 'RandomAccessCollection'}}
+
+  // UTF16View.Collection is random-access with Foundation, bidirectional without
+  acceptsCollection(s.utf16)
+  acceptsBidirectionalCollection(s.utf16)
+  acceptsRandomAccessCollection(s.utf16)
+
+  acceptsCollection(s.unicodeScalars)
   acceptsBidirectionalCollection(s.unicodeScalars)
   acceptsRandomAccessCollection(s.unicodeScalars) // expected-error{{argument type 'String.UnicodeScalarView' does not conform to expected type 'RandomAccessCollection'}}
+
+  acceptsCollection(s.characters)
   acceptsBidirectionalCollection(s.characters)
   acceptsRandomAccessCollection(s.characters) // expected-error{{argument type 'String.CharacterView' does not conform to expected type 'RandomAccessCollection'}}
+}
   
-  // UTF16View.Collection is random-access with Foundation, bidirectional without
-  acceptsRandomAccessCollection(s.utf16)
+// Positive and negative tests for String index types. Testing the complete
+// set of conformance just in case protocol hierarchy changes.
+func acceptsCollectionIndex<I: Comparable>(_: I) {}
+func acceptsBidirectionalCollectionIndex<I: Comparable>(_: I) {}
+func acceptsRandomAccessCollectionIndex<I: Strideable>(_: I) {}
```

Comment:
We have dropped the `Strideable` requirement for `RandomAccessCollection`.

Could you instead change these tests to call `expectCollectionAssociatedTypes` like in `validation-test/stdlib/Set.swift`:

```
SetTestSuite.test("AssociatedTypes") {
  typealias Collection = Set<MinimalHashableValue>
  expectCollectionAssociatedTypes(
    collectionType: Collection.self,
    iteratorType: SetIterator<MinimalHashableValue>.self,
    subSequenceType: Slice<Collection>.self,
    indexType: SetIndex<MinimalHashableValue>.self,
    indexDistanceType: Int.self,
    indicesType: DefaultIndices<Collection>.self)
}
```

Since that requires importing StdlibCollectionUnittest, please move the test to validation.